### PR TITLE
rgwlc:  warn on missing RGW_ATTR_LC

### PR DESCRIPTION
**Pull request by mattbenjamin**
This should not happen.  If it does (e.g., due to damaged bucket_info),
log the event to assist with debugging.

Fixes: https://tracker.ceph.com/issues/53728

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by dang — APPROVED** [review]

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -1695,9 +1695,14 @@ int RGWLC::bucket_lc_process(string& shard_id, LCWorker* worker,
     return -ENOENT;
   }
 
-  map<string, bufferlist>::iterator aiter = bucket->get_attrs().find(RGW_ATTR_LC);
-  if (aiter == bucket->get_attrs().end())
+  map<string, bufferlist>::iterator aiter
+    = bucket->get_attrs().find(RGW_ATTR_LC);
+  if (aiter == bucket->get_attrs().end()) {
+    ldpp_dout(this, 0) << "bucket_attrs.find(RGW_ATTR_LC) failed for "
```
Comment (ivancich) [inline src/rgw/rgw_lc.cc:1701]: Don't know how warning-ish this is, but consider prefixing log message with "WARNING: ".

**Review by ivancich — APPROVED** [review]

File: src/rgw/rgw_lc.cc (hunk 1 above)
Comment (mattbenjamin) [reply src/rgw/rgw_lc.cc:1701]: can do

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
passed qa https://pulpito.ceph.com/cbodley-2022-01-28_14:34:56-rgw-wip-cbodley-testing-distro-default-smithi/

**Comment by cbodley** [comment]
jenkins test docs

**Comment by cbodley** [comment]
jenkins test api

**Pull request merged**